### ISSUE TITLE
Add option to dump command line

### ISSUE THREAD

**Issue by dcbaker** [mesonbuild/meson#4637]
Several people have asked for the ability to get their command line back in a form they can feed back into meson. Since we already store that in a nice form it should be pretty easy to print it back for them.

**Comment by xclaesse** [comment]
Do you have a use-cases for that? Since we have --wipe already I'm not sure why you would want the command line.

**Comment by dcbaker** [comment]
Yes, one of the mesa developers has a lot of different machines he works on (it's a hardware driver thing), and wants to be able to get the configuration back in a form he can copy to another machine. BTW, thanks for doing the --wipe thing.

**Comment by xclaesse** [comment]
Sounds legit to me, shouldn't be too hard to implement.

**Comment by jpakkane** [comment]
Alternatively, if the native file was a bit more fleshed out, we could make this output a native file for later usage. It has the additional niceness that it can be used in other projects directly.

**Comment by dcbaker** [comment]
I actually like the native file idea. We could just replace the internal cmd_line.txt with a native file, which has the added advantage of not allowing us to break the format accidentally

**Comment by xclaesse** [comment]
We don't write configure options in the native/cross files, so that would be a whole new concept in itself.

**Comment by dcbaker** [comment]
@jpakkane suggested it in #4636

**Comment by marc-h38** [comment]
https://github.com/keith-packard/picolibc/tree/80528c684 has a collection of `do-xxx-configure` scripts  that seem to workaround the lack of this.

As pointed by @eli-schwartz on IRC, Autotools have CONFIG_SITE
https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/html_node/Site-Defaults.html
>  options set shell variables that have the same names as the options, with any dashes turned into underscores

**Comment by xclaesse** [comment]
I think we should add something like this to native/cross files:
```
[options]
prefix = '/system'
libdir = 'lib'
some_subproject:some_option = 'some value'
```

Those would be default values that could still be overridden by `-Dprefix=/something`.

Should be easy to implement and that covers the piclibc use-case. I makes a lot of sense to me, it's really common when cross compiling to a given platform to have specific values for options.

**Comment by dcbaker** [comment]
you can already set paths via the `[paths]` section

for other things that's what #3001 is about, which I'm working on.

**Comment by dcbaker** [comment]
@marc-h38 I've been meaning to propose a "default" native file, but I was going to do that after #3001

**Comment by eli-schwartz** [comment]
Yes, having meson automatically load `/usr/share/meson/siteconfig` as a native file would be very nice and a good analogue to autotools' `$CONFIG_SITE` file.

**Comment by germandiagogomez** [comment]
I support this feature of having native files config things. I am having the need for this with so many permutations of configs. It seems that having something in default_options (such as C++ standard) cannot be overriden when using Meson from the command line at first invocation.

**Comment by dcbaker** [comment]
We already have native config files, https://mesonbuild.com/Native-environments.html, this proposal is about serializing all of the arguments on the command line to a native file instead of a one-off format.
